python/ray/_private/worker.py

```diff
@@ -2682,50 +2685,65 @@ def put(
 @PublicAPI
 @client_mode_hook
 def wait(
-    object_refs: List["ray.ObjectRef"],
+    ray_waitables: Union["ObjectRefType[R]", "StreamingObjectRefGeneratorType[R]"],
```

Comment:
Does it make sense to make StreaingObjectRefGenerator generic since it can return multiple objects with different types?

Reply:
Yeah you can do StreaingObjectRefGenerator[Union[...]]

in that case

Reply:
I think this is important decision though. Normally generator has a type annotation Generator[yield type, send type, return type], so I think it makes sense to have it in generator too. But I created a sync meeting with @pcmoritz tomorrow to discuss a bit more details.